.github/workflows/docs.yml

```diff
@@ -8,61 +8,231 @@ on:
+      - name: Create website directory
+        run: |
+          cd /home/runner/work/djl
+          mkdir site
+
+      - name: Set-up Git Config
+        run: |
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+      - name: Checkout DJL Repo
         uses: actions/checkout@v4
-      - name: add mybinder link
+        with: 
+          ref: ${{ github.event.inputs.is-new-version == 'true' && github.event.inputs.djl-tag || 'master' }}
+
+        #This step is necessary because the index.html file in djl/index.html overwrites the index.html file Mike create for the landing page
+        #This is problematic because this index.html corresponds to the djl website rather than the docs website.
+      - name: Delete index.html 
         run: |
-          python3 tools/scripts/add_online_runner.py
-      - name: clone demos
+          rm -f index.html    
+          git add -u 
+          git commit -m "Removed index.html" || echo "No changes to commit"
+        shell: bash
+
+      - name: Clone DJL-Serving
+        env:
+          IS_NEW_VERSION: ${{ github.event.inputs.is-new-version }}
+          DJL_SERVING_TAG: ${{ github.event.inputs.djl-serving-tag }}
         run: |
-          cd docs
+          cd /home/runner/work/djl/djl/docs
+          git clone https://github.com/deepjavalibrary/djl-serving.git serving
+          cd serving
+          if [ "$IS_NEW_VERSION" = "true" ]; then
+            git checkout "$DJL_SERVING_TAG"
+          fi
+        shell: bash
+
+
+      - name: Clone DJL-Demo 
+        env:
+          IS_NEW_VERSION:  ${{ github.event.inputs.is-new-version }}
+          DJL_DEMO_TAG: ${{ github.event.inputs.djl-demo-tag }}
+        run: |
+          cd /home/runner/work/djl/djl/docs
           git clone https://github.com/deepjavalibrary/djl-demo.git demos
-      - name: clone serving
+          cd demos
+          if [ "$IS_NEW_VERSION" = "true" ]; then
+            git checkout "$DJL_DEMO_TAG"
+          fi
+        shell: bash
+        
+      - name: Add mybinder link
         run: |
-          cd docs
-          git clone https://github.com/deepjavalibrary/djl-serving.git serving
+          python3 tools/scripts/add_online_runner.py
+        
       - name: run Notebooks
         run: |
           cd docs/demos/jupyter
           bash test_notebook.sh
+
+      #Phase III: Retrive Current Website
+      #This phase retrieves the current version of the website that is stored in S3. The key pieces are the index.html and versions.json which serve as the landing page and the tracker of previous versions. 
+      #Potential Improvement: See if mike still works if only downloading the index.html file and versions.json rather than the entire previous website. 
+      - name: Sync gh-pages
```

Comment:
Can we just create a shell script for all these steps?, ideally this script can be executed locally